teaching_reproducibility.Rmd

```diff
@@ -20,69 +20,117 @@ library(knitr)
+(i.e., a version controlled project history).
+This history is important for recording how and why
+decisions to use one method or another were made,
+among other things.
+
+The reason we embrace this definition of data science,
+is that we believe that data science work should bring insight
+(e.g., answer an important research question)
+and employ reproducible and auditable methods
+so that trustworthy results and data products can be created.
+Results and data products can be generated
+without reproducible and auditable methods,
+however,
+when they are built this way there is less confidence
+in how the results or products were created.
+This is mainly because of three reasons:
+
+1. They lack evidence that the results or product could be regenerated given the same input computational methods, and conditions.
+2. There is insufficient evidence of the steps taken during creation.
+3. There is an incomplete record of how and why analysis decisions were made.
+
+In addition to contributing to the trustworthiness of data science work,
+employing reproducible and auditable methods and workflows
+bring additional benefits to data scientists,
+such as in the form of more effective collaboration.
+Data science is an inherently collaborative science,
+and the emphasis of reproducible and auditable methods in data science
+greatly facilitates the act of collaborating in many context,
+further emphasizing the importance of learning this skill well.
+
+Although the many benefits of reproducible workflows
+may make them seem like an exciting topic for incoming students,
+the experience when enter a classroom of curious data scientists in training
+is quite the opposite.
+Students are usually keen to learn about data science
+but what they're most excited about is the second part of
+its definition: extracting insights from the data.
+
+Students are often not even aware of the reproducible
+and auditable processes of data science
+and when they first hear about them,
+they tend to regard them as an inconvenient means to an end
+rather than an important skill to master.
+This outlook is likely at least partly motivated by that these processes
+do not directly lead to novel insights
+in the same way as a predictive model might,
+which is what many students have in mind when they envision the work of a data scientist.
+This negative predisposition creates another barrier to overcome
+when teaching the reproducible and auditable aspects of data science.
+
+An additional pedagogical challenge
+is that the tools that we use for reproducibility
+are not necessarily smooth and easy to learn,
+but often have a steep learning curve.
+Over our five years of teaching these topics at UBC
+we've found three pedagogical strategies
+that are particularly effective for teaching reproducibility successfully:
 
 1. placing extra emphasis on motivation
-2. direct instruction
-3. lots of practice 
+2. live demonstration
```

Comment:
I replaced "direct instruction" with "live demonstration" based on your dossier, but we could also consider something like "guided instructions" if we want to talk in broader terms (I think we mostly talk about live demoes here so should be fine).

Reply:
I think guided instruction might be better? And then there are three examples, pre-readings, live demonstrations and worksheets? That would split this section up into three, like the previous? We could elaborate on the pre-readings, live demos and worksheets? The live demos could be likened to live coding, but with additional challenges for demonstrating when you have to use point and click tools (as soon as you click, the users cannot see what you just did...)?